docs/xbcloud-binary-fifo-datasink.md

```diff
@@ -0,0 +1,64 @@
+# FIFO data sink
+
+The feature is in [tech preview](glossary.md#tech-preview).
+
+[Percona XtraBackup 8.0.33-28](release-notes/8.0/8.0.33-28.0.md) implements a new data sink that uses the first in, first out (FIF0) method. With the `FIFO` data sink feature, you can benefit from faster backups by streaming data in parallel from XtraBackup to `xbcloud`.
+
+## Stream to object storage
+
+When taking a backup, you can save the files locally or stream the files to either different servers or object storage. 
+
+Before [Percona XtraBackup 8.0.33-28](release-notes/8.0/8.0.33-28.0.md), XtraBackup streams data to object storage writing to STDOUT (a pipe) and using [`xbcloud`](xbcloud-binary-overview.md) to read from STDIN (standard input device). 
+
+This is how it works:
+
+XtraBackup spawns multiple copy threads and each copy thread reads a data chunk from a specific file. Then each copy thread writes the data chunks to a pipe (STDOUT). An xbcloud read thread reads each chunk of STDIN data. Then xbcloud uploads the chunks to an object storage using an async request, and adds a callback to an event handler list. The xbcloud event handler executes the callback depending on the response from the object storage (success or failure).
+
+![image](_static/backup-streamed-to-object-storage.png)
+
+XtraBackup streaming has a limitation of 1.8 Gbps. Xtrabackup uses multiple copy threads to read data chunks from files in parallel, but the threds are serialized when Xtrabackup writes them to STDOUT. Uploading chunks on a Wide Area Network (WAN) experiences no lag, but streaming backups using a Local Area Network (LAN) can experience a lag because those configurations are designed for at least 10Gbps.
+
+Starting with [Percona XtraBackup 8.0.33-28](release-notes/8.0/8.0.33-28.0.md), XtraBackup can stream data to `xbcloud` using `FIFO` (named pipes).
+
+This is how it works:
+
+XtraBackup spawns multiple copy threads and each copy thread reads a data chunk from a specific file. Then multiple FIFO files are created to store the data from XtraBackup. Each xtrabackup copy thread writes the data chunks to a specific FIFO file. Xbcloud creates multiple pairs of read threads and event handlers to read from FIFO. Then xbcloud uploads data to object storage using async request. The xbcloud event handler executes the callback depending on the response from the object storage (success or failure). 
```

Comment:
the multiple pairs statement is also not clear.  We could simply say,
xbcloud will read from the FIFO streams that xtrabackup created and uploads the..

Reply:
updated